src/components/images/imageLoader.js

```diff
@@ -1,50 +1,83 @@
-define(['lazyLoader', 'imageFetcher', 'layoutManager', 'browser', 'appSettings', 'userSettings', 'require', 'css!./style'], function (lazyLoader, imageFetcher, layoutManager, browser, appSettings, userSettings, require) {
-    'use strict';
+import * as lazyLoader from 'lazyLoader';
+import * as userSettings from 'userSettings';
+import 'css!./style';
+/* eslint-disable indent */
 
-    var self = {};
+    export function lazyImage(elem, source = elem.getAttribute('data-src')) {
+        if (!source) {
+            return;
+        }
 
-    function fillImage(elem, source, enableEffects) {
+        fillImageElement(elem, source);
+    }
 
-        if (!elem) {
-            throw new Error('elem cannot be null');
+    export function fillImage(entry) {
+        if (!entry) {
+            throw new Error('entry cannot be null');
         }
 
-        if (!source) {
-            source = elem.getAttribute('data-src');
+        var source = undefined;
+        if (entry.target) {
+            source = entry.target.getAttribute('data-src');
+        } else {
+            source = entry;
         }
 
-        if (!source) {
-            return;
+        if (entry.intersectionRatio > 0) {
+            fillImageElement(entry.target, source);
```

Comment:
You removed check for `source` - we will fail `if (url === undefined)`. That's why I left `if (source) ...`

Reply:
```suggestion
            if (source) fillImageElement(entry.target, source);
```
You can fill image if you have `source`.

Reply:
Added the suggestion, I can't seem to trigger the load error anymore :+1: